internal/sidecarlogresults/sidecarlogresults.go

```diff
@@ -102,20 +140,39 @@ func LookForResults(w io.Writer, runDir string, resultsDir string, resultNames [
 		resultFile := resultFile
 
 		g.Go(func() error {
-			value, err := os.ReadFile(filepath.Join(resultsDir, resultFile))
-			if os.IsNotExist(err) {
-				return nil
-			} else if err != nil {
-				return fmt.Errorf("error reading the results file %w", err)
+			newResult, err := readResults(resultsDir, resultFile, "", taskResultType)
+			if err != nil {
+				return err
 			}
-			newResult := SidecarLogResult{
-				Name:  resultFile,
-				Value: string(value),
+			if newResult.Name == "" {
+				return nil
 			}
 			results <- newResult
 			return nil
 		})
 	}
+
+	for sName, sresults := range stepResults {
+		sresults := sresults
+		sName := sName
```

Comment:
why do we need this?

Reply:
go func. Async execution. Without it, it always gets the last one in the list.